File: projeto/java/Broker.java
import org.zeromq.ZMQ;

public class Broker {
    public static void main(String[] args) {

        ZMQ.Context context = ZMQ.context(1);

        ZMQ.Socket frontend = context.socket(ZMQ.ROUTER);
        frontend.bind("tcp://*:5555");

        ZMQ.Socket backend = context.socket(ZMQ.DEALER);
        backend.bind("tcp://*:5556");

        ZMQ.proxy(frontend, backend, null);

        frontend.close();
        backend.close();
        context.close();
    }
}

File: projeto/python/broker.py
import zmq

context = zmq.Context()

client_socket = context.socket(zmq.ROUTER)
client_socket.bind("tcp://*:4445")

server_socket = context.socket(zmq.DEALER)
server_socket.bind("tcp://*:4446")

print("Broker iniciado...")

# 🔥 proxy automático
zmq.proxy(client_socket, server_socket)
